src/model.ts

```diff
@@ -516,6 +521,47 @@ export function build<TSchema extends BaseSchema, TOptions extends SchemaOptions
     }
   );
 
+  /**
+   * @description
+   * Performs an optimized `find` to test for the existence of any document matching the filter criteria.
+   *
+   *
+   * @param filter {Filter<TSchema>}
+   * @param [options] {Omit<FindOptions<TSchema>, "projection" | "limit" | "sort" | "skip">}
+   *
+   * @returns {Promise<boolean>}
+   *
+   * @example
+   * const isAlreadyActive = await User.exists({
+   *   firstName: 'John',
+   *   lastName: 'Wick',
+   *   active: true
+   * });
+   */
+  model.exists = async function exists(
+    filter: Filter<TSchema>,
+    options?: Omit<FindOptions<TSchema>, 'projection' | 'limit' | 'sort' | 'skip'>
+  ): Promise<boolean> {
+    // If there are any entries in the filter, we project out the value from
+    // only one of them.  In this way, if there is an index that spans all the
+    // parts of the filter, this can be a "covered" query.
+    // @see https://www.mongodb.com/docs/manual/core/query-optimization/#covered-query
+    //
+    // Note that we must explicitly remove `_id` from the projection; it is often not
+    // present in compound indexes, and mongo will automatically include it in the
+    // result unless you explicitly exclude it from the projection.
+    //
+    // If you don't pass any filter option, we instead project out the primary
+    // key, `_id` (which will override the earlier exclusion).
+    const key = Object.keys(filter)[0] || '_id';
+    const result = await model.findOne(filter, {
+      // @ts-expect-error The `_id` attribute is not found via Mongo's NestedPaths helper?
```

Comment:
This error started coming up after rebasing, it seems like something changed in Mongo's `NestedPaths` generic to make this not work as expected?

It's rather confusing, as the error looks like this:
```
src/model.ts:558:21 - error TS2322: Type '{ [x: string]: number; _id: 0; }' is not assignable to type 'Partial<Record<Join<NestedPaths<WithId<TSchema>>, ".">, number>>'.
  Object literal may only specify known properties, and '_id' does not exist in type 'Partial<Record<Join<NestedPaths<WithId<TSchema>>, ".">, number>>'.

558       projection: { _id: 0, [key]: 1 },
                        ~~~~~~
```

The `WithId` helper in that chain seems like it should ensure that the `_id` attribute is present, so the fact that it is missing in the final result is extremely surprising to me.

I spent some time trying to figure out what changed in there, and noticed that there is a pending revert of some changes in this area: https://github.com/mongodb/node-mongodb-native/commit/7f94f0a19d48cc3e779216a5b4c51881ab8ddbf0

That seems likely to resolve this error (when a new Mongo version with that revert is released and the dependency bumped), but analyzing the types there is extremely difficult for me. Any thoughts or suggestions on better approaches than this `ts-expect-error`?

Reply:
IMO the `expect-error` is acceptable here for now.

Reply:
@noahsilas The revert you linked to from `mongodb` is already shipped in `mongodb` v4.9.1. Are you sure you were on the latest version?

Either way, we can add a `@ts-expect-error` here to get this PR ready.